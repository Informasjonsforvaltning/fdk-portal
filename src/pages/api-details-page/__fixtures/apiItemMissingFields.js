export const apiItemMissingFields = {
  id: 'c541c305-4e4f-43ed-be3e-5434740ec1b5',
  uri: 'jar:file:/app.jar!/BOOT-INF/classes!/datakatalog-api.json',
  apiSpecification: {
    info: {
      description:
        'Provides a basic search api against the National Data Directory of Norway',
      version: '1.0',
      title: 'National Data Directory Search API',
      termsOfService: 'https://fellesdatakatalog.brreg.no/about',
      contact: {
        name: 'Brønnøysundregistrene',
        url: 'https://fellesdatakatalog.brreg.no',
        email: 'fellesdatakatalog@brreg.no'
      },
      license: {
        name: 'License of API',
        url: 'http://data.norge.no/nlod/no/2.0'
      }
    }
  }
};
