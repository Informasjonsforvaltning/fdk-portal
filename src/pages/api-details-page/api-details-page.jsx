import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import DocumentMeta from 'react-document-meta';

import localization from '../../lib/localization';
import { getTranslateText } from '../../lib/translateText';
import { HarvestDate } from '../../components/harvest-date/harvest-date.component';
import { SearchHitHeader } from '../../components/search-hit-header/search-hit-header.component';
import ApiUsageInstruction from './api-usage-instruction/api-usage-instruction.component';
import { ApiEndpoints } from './api-endpoints/api-endpoints.component';
import { ShowMore } from '../../components/show-more/show-more';
import { StickyMenu } from '../../components/sticky-menu/sticky-menu.component';
import { ListRegular } from '../../components/list-regular/list-regular.component';
import { TwoColRow } from '../../components/list-regular/twoColRow/twoColRow';
import { DatasetReference } from './dataset-reference/dataset-reference.component';
import { InformationModelReference } from './informationmodel-reference/informationmodel-reference.component';
import { convertToSanitizedHtml } from '../../lib/markdown-converter';
import './api-details-page.scss';
import { AlertMessage } from '../../components/alert-message/alert-message.component';
import {
  iconIsFree,
  iconIsNotFree,
  iconIsNotOpenAccess,
  iconIsNotOpenLicense,
  iconIsOpenAccess,
  iconIsOpenLicense
} from '../../components/api-icons';
import {
  getReferenceDataByCode,
  REFERENCEDATA_PATH_APISERVICETYPE
} from '../../redux/modules/referenceData';

const renderDescription = descriptionFormatted => {
  if (!descriptionFormatted) {
    return null;
  }

  return (
    <ShowMore showMoreButtonText={localization.showFullDescription}>
      <div
        dangerouslySetInnerHTML={{
          __html: convertToSanitizedHtml(descriptionFormatted)
        }}
      />
    </ShowMore>
  );
};

const renderFormats = formats => {
  if (!formats || formats.length === 0) {
    return null;
  }
  const formatItems = formats =>
    formats.map((item, index) => (
      <span key={index}>
        {index > 0 ? ', ' : ''}
        {item}
      </span>
    ));
  return (
    <ListRegular title={localization.format}>
      <div className="d-flex list-regular--item">{formatItems(formats)}</div>
    </ListRegular>
  );
};

const renderApiUsageInstruction = (servers, apiSpecUrl, apiDocUrl) =>
  servers &&
  servers.length && (
    <ApiUsageInstruction
      servers={servers}
      apiSpecUrl={apiSpecUrl}
      apiDocUrl={apiDocUrl}
    />
  );

const renderApiEndpoints = (paths, apiSpecUrl, apiDocUrl) => {
  if (!paths) {
    return null;
  }
  return (
    <ApiEndpoints
      name={localization.api.endpoints.operations}
      paths={paths}
      apiSpecUrl={apiSpecUrl}
      apiDocUrl={apiDocUrl}
    />
  );
};

const renderAPIInfo = ({ children }) => {
  if (!children) {
    return null;
  }
  return <ListRegular title={localization.apiInfo}>{children}</ListRegular>;
};

const renderDatasetReferences = references => {
  if (!references || (references && references.length === 0)) {
    return null;
  }
  const children = items =>
    items.map((item, index) => (
      <DatasetReference
        key={`${index}-${item.id}`}
        datasetReference={item}
        index={index}
      />
    ));
  return (
    <ListRegular title={localization.datasetReferences}>
      {children(references)}
    </ListRegular>
  );
};

const renderInformationModelReferences = informationModels => {
  if (
    !Array.isArray(informationModels) ||
    (informationModels && informationModels.length === 0)
  ) {
    return null;
  }
  const children = items =>
    items.map((item, index) => (
      <InformationModelReference
        key={`${index}-${item.id}`}
        informationModelReference={item}
        index={index}
      />
    ));
  return (
    <ListRegular title={localization.informationModelReferences}>
      {children(informationModels)}
    </ListRegular>
  );
};

const getContactPointKey = contactPoint =>
  contactPoint &&
  (contactPoint.uri ||
    contactPoint.organizationName ||
    contactPoint.email ||
    contactPoint.phone);

const renderContactPoint = contactPoint => {
  const { uri, organizationName, email, phone } = contactPoint;
  return (
    <React.Fragment key={getContactPointKey(contactPoint)}>
      {(uri || organizationName) && (
        <TwoColRow
          col1={localization.contactPoint}
          col2={
            uri ? (
              <a href={uri}>
                {organizationName || uri}
                <i className="fa fa-external-link fdk-fa-right" />
              </a>
            ) : (
              organizationName
            )
          }
        />
      )}
      {email && (
        <TwoColRow
          col1={localization.email}
          col2={
            <a title={email} href={`mailto:${email}`} rel="noopener noreferrer">
              {email}
            </a>
          }
        />
      )}
      {phone && <TwoColRow col1={localization.phone} col2={phone} />}
    </React.Fragment>
  );
};

const renderContactPoints = contactPoints => {
  if (
    !contactPoints ||
    !Array.isArray(contactPoints) ||
    contactPoints.length < 1 ||
    !contactPoints.some(getContactPointKey)
  ) {
    return null;
  }

  const children = items => items.map(item => renderContactPoint(item));

  return (
    <ListRegular title={localization.contactInfo}>
      {children(contactPoints)}
    </ListRegular>
  );
};

const renderTermsAndRestrictions = (
  termsOfService,
  license,
  cost,
  usageLimitation,
  performance,
  availability
) => {
  if (
    !(
      termsOfService ||
      license ||
      cost ||
      usageLimitation ||
      performance ||
      availability
    )
  ) {
    return null;
  }

  return (
    <ListRegular
      title={localization.api.termsAndRestrictions.termsAndRestrictions}
    >
      {termsOfService && (
        <TwoColRow
          col1={localization.api.termsAndRestrictions.termsOfService}
          col2={
            <a href={termsOfService}>
              {termsOfService}
              <i className="fa fa-external-link fdk-fa-right" />
            </a>
          }
        />
      )}
      {license && (
        <TwoColRow
          col1={localization.api.termsAndRestrictions.license}
          col2={
            <a href={_.get(license, 'url')}>
              {_.get(license, 'name')}
              <i className="fa fa-external-link fdk-fa-right" />
            </a>
          }
        />
      )}
      {cost && (
        <TwoColRow
          col1={localization.api.termsAndRestrictions.cost}
          col2={cost}
        />
      )}
      {usageLimitation && (
        <TwoColRow
          col1={localization.api.termsAndRestrictions.usageLimitation}
          col2={usageLimitation}
        />
      )}
      {performance && (
        <TwoColRow
          col1={localization.api.termsAndRestrictions.performance}
          col2={performance}
        />
      )}
      {availability && (
        <TwoColRow
          col1={localization.api.termsAndRestrictions.availability}
          col2={availability}
        />
      )}
    </ListRegular>
  );
};

const renderStickyMenu = (apiItem, informationModels) => {
  const menuItems = [];
  if (_.get(apiItem, 'description')) {
    menuItems.push({
      name: getTranslateText(_.get(apiItem, 'title')),
      prefLabel: localization.description
    });
  }
  if (_.get(apiItem, 'formats', []).length > 0) {
    menuItems.push({
      name: localization.format,
      prefLabel: localization.format
    });
  }
  if (_.get(apiItem, ['apiSpecification', 'servers'], []).length) {
    menuItems.push({
      name: localization.api.servers.title,
      prefLabel: localization.api.servers.title
    });
  }
  if (_.get(apiItem, ['apiSpecification', 'paths'])) {
    menuItems.push({
      name: localization.api.endpoints.operations,
      prefLabel: localization.api.endpoints.operations
    });
  }
  if (_.get(apiItem, 'serviceType')) {
    menuItems.push({
      name: localization.serviceType,
      prefLabel: localization.serviceType
    });
  }
  if (
    _.get(apiItem, ['apiSpecification', 'info', 'termsOfService']) ||
    _.get(apiItem, ['apiSpecification', 'info', 'license']) ||
    _.get(apiItem, 'cost') ||
    _.get(apiItem, 'usageLimitation') ||
    _.get(apiItem, 'performance') ||
    _.get(apiItem, 'availability')
  ) {
    menuItems.push({
      name: localization.api.termsAndRestrictions.termsAndRestrictions,
      prefLabel: localization.api.termsAndRestrictions.termsAndRestrictions
    });
  }
  if (_.get(apiItem, 'datasetReferences')) {
    menuItems.push({
      name: localization.datasetReferences,
      prefLabel: localization.datasetReferences
    });
  }
  if (Array.isArray(informationModels) && informationModels.length > 0) {
    menuItems.push({
      name: localization.informationModelReferences,
      prefLabel: localization.informationModelReferences
    });
  }
  if (_.get(apiItem, 'contactPoint')) {
    menuItems.push({
      name: localization.contactInfo,
      prefLabel: localization.contactInfo
    });
  }
  return <StickyMenu title={localization.goTo} menuItems={menuItems} />;
};

const renderServiceType = (serviceType, referenceData) => {
  if (!serviceType) {
    return null;
  }
  const referenceDataServiceTypeItem = getReferenceDataByCode(
    referenceData,
    REFERENCEDATA_PATH_APISERVICETYPE,
    serviceType
  );
  return (
    <ListRegular title={localization.serviceType}>
      <TwoColRow
        col1={localization.serviceType}
        col2={getTranslateText(
          _.get(referenceDataServiceTypeItem, 'prefLabel')
        )}
      />
    </ListRegular>
  );
};

const renderExpiredOrDeprecatedVersion = item => {
  const statusCode = _.get(item, 'statusCode');
  const deprecationInfoExpirationDate = _.get(
    item,
    'deprecationInfoExpirationDate'
  );
  const deprecationInfoReplacedWithUrl = _.get(
    item,
    'deprecationInfoReplacedWithUrl'
  );
  if (statusCode === 'REMOVED') {
    return (
      <AlertMessage type="info">
        <span>{localization.statusRemoved} </span>
        {deprecationInfoReplacedWithUrl && (
          <a href={deprecationInfoReplacedWithUrl}>
            {localization.statusReplacedUrl}
          </a>
        )}
      </AlertMessage>
    );
  }
  if (statusCode === 'DEPRECATED') {
    return (
      <AlertMessage type="info">
        <span>
          {deprecationInfoExpirationDate &&
            localization.formatString(
              localization.statusDeprecated,
              localization.during,
              deprecationInfoExpirationDate.substring(0, 4)
            )}
          {!deprecationInfoExpirationDate &&
            localization.formatString(localization.statusDeprecated, '', '')}
        </span>

        {deprecationInfoReplacedWithUrl && (
          <a href={deprecationInfoReplacedWithUrl}>
            {localization.statusReplacedUrl}
          </a>
        )}
      </AlertMessage>
    );
  }
  return null;
};

export const ApiDetailsPage = ({
  apiItem,
  publisherItems,
  referencedDatasets,
  referencedInformationModels,
  referenceData,
  fetchPublishersIfNeeded,
  fetchApiStatusIfNeeded,
  fetchApiServiceTypeIfNeeded
}) => {
  if (!apiItem) {
    return null;
  }

  fetchPublishersIfNeeded();
  fetchApiStatusIfNeeded();
  fetchApiServiceTypeIfNeeded();

  const internalApiSpecUrl = `/api/apis/${apiItem.id}/spec`;

  const meta = {
    title: getTranslateText(apiItem.title),
    description: getTranslateText(apiItem.description)
  };

  const { isFree, isOpenAccess, isOpenLicense } = apiItem;

  return (
    <main id="content" className="container">
      <article>
        <div className="row">
          <div className="col-12 col-lg-8 offset-lg-4">
            <DocumentMeta {...meta} />

            <div className="d-flex fdk-detail-date mb-5">
              <i className="align-self-center fdk-icon-catalog-api mr-2" />
              <strong className="align-self-center">
                {localization.api.apiDescription}&nbsp;
              </strong>
              <HarvestDate
                className="align-self-center"
                harvest={apiItem.harvest}
              />
            </div>
          </div>
        </div>

        <div className="row">
          <div className="col-12 col-lg-4 ">
            {renderStickyMenu(apiItem, referencedInformationModels)}
          </div>

          <section className="col-12 col-lg-8 mt-3 api-details-section">
            <div name={getTranslateText(apiItem.title)}>
              <SearchHitHeader
                title={getTranslateText(apiItem.title)}
                publisherLabel={`${localization.provider}:`}
                publisher={apiItem.publisher}
                publisherItems={publisherItems}
                nationalComponent={apiItem.nationalComponent}
                statusCode={apiItem.statusCode}
                referenceData={referenceData}
                darkThemeBackground
              />
              {renderExpiredOrDeprecatedVersion(apiItem)}

              {renderDescription(
                apiItem.descriptionFormatted || apiItem.description
              )}
              <div className="access-icons mb-5 d-flex justify-content-between">
                {isFree === true && iconIsFree()}
                {isFree === false && iconIsNotFree()}
                {isOpenAccess === true && iconIsOpenAccess()}
                {isOpenAccess === false && iconIsNotOpenAccess()}
                {isOpenLicense === true && iconIsOpenLicense()}
                {isOpenLicense === false && iconIsNotOpenLicense()}
              </div>
            </div>

            {renderFormats(apiItem.formats)}

            {renderApiUsageInstruction(
              _.get(apiItem, ['apiSpecification', 'servers'], []),
              apiItem.apiSpecUrl || internalApiSpecUrl,
              apiItem.apiDocUrl
            )}

            {renderApiEndpoints(
              apiItem.apiSpecification && apiItem.apiSpecification.paths
            )}

            {renderAPIInfo({})}

            {renderTermsAndRestrictions(
              _.get(apiItem, ['apiSpecification', 'info', 'termsOfService']),
              _.get(apiItem, ['apiSpecification', 'info', 'license']),
              _.get(apiItem, 'cost'),
              _.get(apiItem, 'usageLimitation'),
              _.get(apiItem, 'performance'),
              _.get(apiItem, 'availability')
            )}

            {renderServiceType(_.get(apiItem, 'serviceType'), referenceData)}

            {renderDatasetReferences(referencedDatasets)}

            {renderInformationModelReferences(referencedInformationModels)}

            {renderContactPoints(apiItem.contactPoint)}
            <div style={{ height: '75vh' }} />
          </section>
        </div>
      </article>
    </main>
  );
};

ApiDetailsPage.defaultProps = {
  apiItem: null,
  publisherItems: null,
  referenceData: null,
  fetchPublishersIfNeeded: () => {},
  fetchApiStatusIfNeeded: _.noop,
  fetchApiServiceTypeIfNeeded: _.noop,
  referencedDatasets: []
};

ApiDetailsPage.propTypes = {
  apiItem: PropTypes.object,
  publisherItems: PropTypes.object,
  referenceData: PropTypes.object,
  fetchPublishersIfNeeded: PropTypes.func,
  fetchApiStatusIfNeeded: PropTypes.func,
  fetchApiServiceTypeIfNeeded: PropTypes.func,
  referencedDatasets: PropTypes.array
};
