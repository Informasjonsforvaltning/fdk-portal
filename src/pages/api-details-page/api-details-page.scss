@import '../../../node_modules/designsystemet/fdk-designsystem-bootstrap4/scss/colors.scss';

.fdk-icon-catalog-api {
    background-image: url("/img/icon-catalog-api-lg.svg");
    width: 3.5em; 
    height: 3.5em;
    display: inline-block;
}

.api-details-section {
    .access-icon {
        background-color: $color-white;;
        border-radius: 5px;
        border: 1px $color-white;;
        padding: 1em 2em;
        margin-right: 0.5em;
    }

    .access-icon:last-child {
        margin-right: 0;
    }
}
