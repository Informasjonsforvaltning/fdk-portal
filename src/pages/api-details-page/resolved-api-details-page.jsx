import _ from 'lodash';
import first from 'lodash/first';
import { resolve } from 'react-resolver';
import { ApiDetailsPage } from './api-details-page';
import { getApi } from '../../api/apis';
import {
  datasetsSearch,
  extractDatasets,
  getDatasetByURI
} from '../../api/datasets';
import {
  extractInformationmodels,
  informationmodelsSearch
} from '../../api/informationmodels';

const getInformationModelByHarvestSourceUri = harvestSourceUri =>
  informationmodelsSearch({ harvestSourceUri })
    .then(extractInformationmodels)
    .then(first);

const memoizedGetApi = _.memoize(getApi);
const memoizedGetDatasetByURI = _.memoize(getDatasetByURI);
const memoizedGetInformationModelByHarvestSourceUri = _.memoize(
  getInformationModelByHarvestSourceUri
);

// todo when we migrate to dcat 2.0, we can have more reasonably link together datasets and apis,
//  right now it is api id stored in that peculiar property.
const getDatasetsByApiId = id =>
  datasetsSearch({
    distributionAccessServiceEndpointDescriptionUri: id,
    returnFields: 'id,title,publisher,description'
  }).then(extractDatasets);
const memoizedGetDatasetsByApiId = _.memoize(getDatasetsByApiId);

const mapProps = {
  apiItem: props => memoizedGetApi(props.match.params.id),
  referencedDatasets: async props => {
    const id = props.match.params.id;
    const getApi = await memoizedGetApi(id);
    const urlArray = _.get(getApi, 'datasetReferences', []).map(
      item => item.uri
    );

    return (await Promise.all(urlArray.map(memoizedGetDatasetByURI)))
      .concat(await memoizedGetDatasetsByApiId(id))
      .filter(Boolean);
  },
  referencedInformationModels: async props => {
    const apiItem = await memoizedGetApi(props.match.params.id);

    const harvestSourceUri = _.get(apiItem, 'harvestSourceUri');

    const informationmodel = await memoizedGetInformationModelByHarvestSourceUri(
      harvestSourceUri
    );

    // this method returns list of referenced information models, but our current api gives one
    return informationmodel ? [informationmodel] : [];
  }
};

export const ResolvedApiDetailsPage = resolve(mapProps)(ApiDetailsPage);
