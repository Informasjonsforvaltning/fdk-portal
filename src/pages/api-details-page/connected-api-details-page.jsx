import { connect } from 'react-redux';
import { fetchPublishersIfNeededAction } from '../../redux/modules/publishers';
import { ResolvedApiDetailsPage } from './resolved-api-details-page';
import {
  fetchReferenceDataIfNeededAction,
  REFERENCEDATA_PATH_APISTATUS,
  REFERENCEDATA_PATH_APISERVICETYPE
} from '../../redux/modules/referenceData';

const mapStateToProps = ({ publishers, referenceData }) => {
  const { publisherItems } = publishers || {
    publisherItems: null
  };

  return {
    publisherItems,
    referenceData
  };
};

const mapDispatchToProps = dispatch => ({
  fetchPublishersIfNeeded: () => dispatch(fetchPublishersIfNeededAction()),
  fetchApiStatusIfNeeded: () =>
    dispatch(fetchReferenceDataIfNeededAction(REFERENCEDATA_PATH_APISTATUS)),
  fetchApiServiceTypeIfNeeded: () =>
    dispatch(
      fetchReferenceDataIfNeededAction(REFERENCEDATA_PATH_APISERVICETYPE)
    )
});

export const ConnectedApiDetailsPage = connect(
  mapStateToProps,
  mapDispatchToProps
)(ResolvedApiDetailsPage);
