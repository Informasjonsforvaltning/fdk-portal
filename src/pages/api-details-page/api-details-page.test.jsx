import React from 'react';
import { shallow } from 'enzyme';
import { ApiDetailsPage } from './api-details-page';
import apiItemComplete from '../../mock/api.response.json';
import { apiItemMissingFields } from './__fixtures/apiItemMissingFields';

test('should render ApiDetailsPage correctly with no apiItem', () => {
  const wrapper = shallow(<ApiDetailsPage />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ApiDetailsPage correctly with apiItem', () => {
  const wrapper = shallow(
    <ApiDetailsPage apiItem={apiItemComplete} isFetchingDataset={false} />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should render ApiDetailsPage correctly with missing api-properties', () => {
  const wrapper = shallow(
    <ApiDetailsPage apiItem={apiItemMissingFields} isFetchingDataset={false} />
  );
  expect(wrapper).toMatchSnapshot();
});
