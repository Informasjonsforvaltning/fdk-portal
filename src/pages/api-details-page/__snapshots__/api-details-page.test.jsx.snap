// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ApiDetailsPage correctly with apiItem 1`] = `
<main
  className="container"
  id="content"
>
  <article>
    <div
      className="row"
    >
      <div
        className="col-12 col-lg-8 offset-lg-4"
      >
        <SideEffect(DocumentMeta)
          description="Provides a basic search api against the National API Directory of Norway"
          title="National API Directory Search API"
        />
        <div
          className="d-flex fdk-detail-date mb-5"
        >
          <i
            className="align-self-center fdk-icon-catalog-api mr-2"
          />
          <strong
            className="align-self-center"
          >
            API-description
             
          </strong>
          <HarvestDate
            className="align-self-center"
            harvest={
              Object {
                "changed": Array [
                  "2018-12-21T09:27:29.577Z",
                  "2019-01-07T16:53:37.588Z",
                  "2019-01-07T16:54:19.940Z",
                  "2019-02-05T11:08:08.046Z",
                  "2019-02-05T11:08:24.889Z",
                  "2019-02-25T14:18:27.207Z",
                  "2019-02-26T08:39:00.913Z",
                  "2019-02-26T08:39:05.880Z",
                  "2019-03-01T17:00:01.476Z",
                  "2019-03-07T17:00:02.358Z",
                  "2019-04-02T10:00:01.620Z",
                  "2019-04-03T04:00:02.084Z",
                  "2019-04-23T10:00:01.993Z",
                  "2019-06-25T04:00:03.176Z",
                  "2019-08-04T04:00:04.028Z",
                ],
                "firstHarvested": "2018-12-19T09:52:51.883Z",
                "lastHarvested": "2019-08-05T10:10:00.051Z",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12 col-lg-4 "
      >
        <StickyMenu
          menuItems={
            Array [
              Object {
                "name": "National API Directory Search API",
                "prefLabel": "Description",
              },
              Object {
                "name": "Format",
                "prefLabel": "Format",
              },
              Object {
                "name": "Start using the API",
                "prefLabel": "Start using the API",
              },
              Object {
                "name": "Operations",
                "prefLabel": "Operations",
              },
              Object {
                "name": "Terms and restrictions",
                "prefLabel": "Terms and restrictions",
              },
              Object {
                "name": "Dataset references",
                "prefLabel": "Dataset references",
              },
              Object {
                "name": "Contact information",
                "prefLabel": "Contact information",
              },
            ]
          }
          title="Go to"
        />
      </div>
      <section
        className="col-12 col-lg-8 mt-3 api-details-section"
      >
        <div
          name="National API Directory Search API"
        >
          <SearchHitHeader
            darkThemeBackground={true}
            externalLink={false}
            nationalComponent={false}
            publisher={
              Object {
                "id": "974760673",
                "name": "REGISTERENHETEN I BRØNNØYSUND",
                "orgPath": "/STAT/912660680/974760673",
                "prefLabel": Object {
                  "no": "Brønnøysundregistrene",
                },
                "uri": "http://data.brreg.no/enhetsregisteret/enhet/974760673",
              }
            }
            publisherItems={null}
            publisherLabel="Provider:"
            publisherTag="span"
            referenceData={null}
            statusCode={null}
            tag="h1"
            theme={Array []}
            title="National API Directory Search API"
            titleLink={null}
          />
          <withStateHandlers(ShowMorePure)
            showMoreButtonText="Show the full description"
          >
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p>Provides a basic search api against the National API Directory of Norway</p>",
                }
              }
            />
          </withStateHandlers(ShowMorePure)>
          <div
            className="access-icons mb-5 d-flex justify-content-between"
          >
            <span
              className="access-icon flex-fill fdk-color-success"
            >
              <span
                className="icon2-icon-api-cost-none"
              >
                <span
                  className="path1"
                />
                <span
                  className="path2"
                />
                <span
                  className="path3"
                />
                <span
                  className="path4"
                />
                <span
                  className="path5"
                />
              </span>
              Free
            </span>
            <span
              className="access-icon flex-fill fdk-color-success"
            >
              <span
                className="icon2-icon-api-access-all"
              >
                <span
                  className="path1"
                />
                <span
                  className="path2"
                />
                <span
                  className="path3"
                />
                <span
                  className="path4"
                />
              </span>
              Public access
            </span>
            <span
              className="access-icon flex-fill fdk-color-success"
            >
              <span
                className="icon2-icon-api-license-open"
              >
                <span
                  className="path1"
                />
                <span
                  className="path2"
                />
                <span
                  className="path3"
                />
                <span
                  className="path4"
                />
                <span
                  className="path5"
                />
                <span
                  className="path6"
                />
              </span>
              Open license
            </span>
          </div>
        </div>
        <ListRegular
          bottomMargin={true}
          title="Format"
        >
          <div
            className="d-flex list-regular--item"
          >
            <span
              key="0"
            >
              application/json
            </span>
          </div>
        </ListRegular>
        <ApiServer
          apiDocUrl="https://informasjonsforvaltning.github.io/felles-datakatalog/api-katalog/api/"
          apiSpecUrl="https://raw.githubusercontent.com/brreg/openAPI/master/specs/api-cat.json"
          servers={
            Array [
              Object {
                "description": "The production API server",
                "url": "https://fellesdatakatalog.brreg.no/api-cat/apis",
              },
            ]
          }
        />
        <ApiEndpoints
          name="Operations"
          paths={
            Object {
              "/": Object {
                "get": Object {
                  "description": "So far only simple queries is supported",
                  "summary": "Queries the api catalog for api specifications",
                },
              },
              "/endpoints": Object {
                "get": Object {
                  "description": "dedicated path to serve endpoints based on organization and/or standard",
                  "summary": "Queries the api catalog for api specifications",
                },
              },
              "/{id}": Object {
                "get": Object {
                  "summary": "Get a specific api",
                },
              },
              "/{id}/schema": Object {
                "get": Object {
                  "summary": "Get a specific api's specification",
                },
              },
            }
          }
        />
        <ListRegular
          bottomMargin={true}
          title="Terms and restrictions"
        >
          <TwoColRow
            col1="Terms of service"
            col1Width={null}
            col2={
              <a
                href="https://fellesdatakatalog.brreg.no/about"
              >
                https://fellesdatakatalog.brreg.no/about
                <i
                  className="fa fa-external-link fdk-fa-right"
                />
              </a>
            }
            col2Width={null}
          />
          <TwoColRow
            col1="License"
            col1Width={null}
            col2={
              <a
                href="http://data.norge.no/nlod/no/2.0"
              >
                License of API
                <i
                  className="fa fa-external-link fdk-fa-right"
                />
              </a>
            }
            col2Width={null}
          />
          <TwoColRow
            col1="Price"
            col1Width={null}
            col2="Gratis"
            col2Width={null}
          />
        </ListRegular>
        <ListRegular
          bottomMargin={true}
          title="Contact information"
        >
          <TwoColRow
            col1="Contact point"
            col1Width={null}
            col2={
              <a
                href="https://fellesdatakatalog.brreg.no"
              >
                Brønnøysundregistrene
                <i
                  className="fa fa-external-link fdk-fa-right"
                />
              </a>
            }
            col2Width={null}
          />
          <TwoColRow
            col1="Email"
            col1Width={null}
            col2={
              <a
                href="mailto:fellesdatakatalog@brreg.no"
                rel="noopener noreferrer"
                title="fellesdatakatalog@brreg.no"
              >
                fellesdatakatalog@brreg.no
              </a>
            }
            col2Width={null}
          />
        </ListRegular>
        <div
          style={
            Object {
              "height": "75vh",
            }
          }
        />
      </section>
    </div>
  </article>
</main>
`;

exports[`should render ApiDetailsPage correctly with missing api-properties 1`] = `
<main
  className="container"
  id="content"
>
  <article>
    <div
      className="row"
    >
      <div
        className="col-12 col-lg-8 offset-lg-4"
      >
        <SideEffect(DocumentMeta)
          description={null}
          title={null}
        />
        <div
          className="d-flex fdk-detail-date mb-5"
        >
          <i
            className="align-self-center fdk-icon-catalog-api mr-2"
          />
          <strong
            className="align-self-center"
          >
            API-description
             
          </strong>
          <HarvestDate
            className="align-self-center"
            harvest={null}
          />
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12 col-lg-4 "
      >
        <StickyMenu
          menuItems={
            Array [
              Object {
                "name": "Terms and restrictions",
                "prefLabel": "Terms and restrictions",
              },
            ]
          }
          title="Go to"
        />
      </div>
      <section
        className="col-12 col-lg-8 mt-3 api-details-section"
      >
        <div
          name={null}
        >
          <SearchHitHeader
            darkThemeBackground={true}
            externalLink={false}
            nationalComponent={false}
            publisher={null}
            publisherItems={null}
            publisherLabel="Provider:"
            publisherTag="span"
            referenceData={null}
            statusCode={null}
            tag="h1"
            theme={Array []}
            title={null}
            titleLink={null}
          />
          <div
            className="access-icons mb-5 d-flex justify-content-between"
          />
        </div>
        0
        <ListRegular
          bottomMargin={true}
          title="Terms and restrictions"
        >
          <TwoColRow
            col1="Terms of service"
            col1Width={null}
            col2={
              <a
                href="https://fellesdatakatalog.brreg.no/about"
              >
                https://fellesdatakatalog.brreg.no/about
                <i
                  className="fa fa-external-link fdk-fa-right"
                />
              </a>
            }
            col2Width={null}
          />
          <TwoColRow
            col1="License"
            col1Width={null}
            col2={
              <a
                href="http://data.norge.no/nlod/no/2.0"
              >
                License of API
                <i
                  className="fa fa-external-link fdk-fa-right"
                />
              </a>
            }
            col2Width={null}
          />
        </ListRegular>
        <div
          style={
            Object {
              "height": "75vh",
            }
          }
        />
      </section>
    </div>
  </article>
</main>
`;

exports[`should render ApiDetailsPage correctly with no apiItem 1`] = `""`;
